base_attachment_object_storage/models/ir_attachment.py

```diff
@@ -0,0 +1,298 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import hashlib
+import inspect
+import logging
+import os
+import psycopg2
+import uuid
+
+from contextlib import contextmanager
+from openerp.tools.translate import _
+from openerp.osv import osv
+from openerp.osv.orm import except_orm
+from openerp import SUPERUSER_ID
+
+
+_logger = logging.getLogger(__name__)
+
+
+def clean_fs(files):
+    _logger.info('cleaning old files from filestore')
+    for full_path in files:
+        if os.path.exists(full_path):
+            try:
+                os.unlink(full_path)
+            except OSError:
+                _logger.info(
+                    "_file_delete could not unlink %s",
+                    full_path, exc_info=True
+                )
+            except IOError:
+                # Harmless and needed for race conditions
+                _logger.info(
+                    "_file_delete could not unlink %s",
+                    full_path, exc_info=True
+                )
+
+
+@contextmanager
+def savepoint(cursor):
+    name = uuid.uuid1().hex
+    cursor.execute('SAVEPOINT "%s"' % name)
+    try:
+        yield
```

Comment:
we didn't need to add the expression/value to yield after one `yield` ? [https://docs.python.org/3/reference/simple_stmts.html#yield](https://docs.python.org/3/reference/simple_stmts.html#yield)

Reply:
It's not mandatory as explained here : https://docs.python.org/3/reference/expressions.html#yieldexpr
But be careful as the "official" use of `yield` is to make generators, in the following doc, you can see how it is used as a context manager (to be use in `with` statements) and a few examples of `yield` usage without value : https://docs.python.org/3/library/contextlib.html

Reply:
@Tonow-c2c There is no value to yield here. 
About @p-tombez 's remark: the context manager in fact expects and uses a generator that yields only one item.